frontend/app-development/contexts/PreviewContext/PreviewContext.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { createContext, useContext, useState } from 'react';
+
+export type PreviewContextProps = {
+  shouldReloadPreview: boolean;
+  doReloadPreview: () => void;
+  previewHasLoaded: () => void;
+};
+
+export const PreviewContext = createContext<Partial<PreviewContextProps>>(undefined);
+
+export type PreviewContextProviderProps = {
+  children: React.ReactNode;
+};
+
+export const PreviewContextProvider = ({ children }: Partial<PreviewContextProviderProps>) => {
+  const [shouldReloadPreview, setShouldReloadPreview] = useState<boolean>(false);
+
+  const doReloadPreview = () => {
```

Comment:
Should we add some tests for the line code-cov is complaining about? 😊

Reply:
To achieve 100% coverage, we can add the following test:

```ts
  it('should doReload and previewHasLoaded should toggle the shouldReloadPreview between true and false', async () => {
    const { result } = renderHook(() => usePreviewContext(), {
      wrapper: PreviewContextProvider,
    });

    const { shouldReloadPreview, doReloadPreview, previewHasLoaded } = result.current;

    expect(shouldReloadPreview).toBe(false);

    act(() => {
      doReloadPreview();
    });

    await waitFor(() => {
      expect(result.current.shouldReloadPreview).toBe(true);
    });

    // Simulate preview loading
    act(() => {
      previewHasLoaded();
    });

    await waitFor(() => {
      expect(result.current.shouldReloadPreview).toBe(false);
    });
  });
```

Reply:
Amazing! Thank you 😄